internal/daemon/vendor-specific-plugins/marvell/main.go

```diff
@@ -350,27 +526,39 @@ func (vsp *mrvlVspServer) Serve(listener net.Listener) error {
 	return err
 }
 
+func (vsp *mrvlVspServer) Stop() {
+	vsp.grpcServer.Stop()
+	vsp.done <- nil
+	vsp.startedWg.Wait()
+}
+func WithPathManager(pathManager utils.PathManager) func(*mrvlVspServer) {
+	return func(vsp *mrvlVspServer) {
+		vsp.pathManager = pathManager
+	}
+}
+
 func NewMarvellVspServer(opts ...func(*mrvlVspServer)) *mrvlVspServer {
-	vsp := &mrvlVspServer{
-		log: ctrl.Log.WithName("MarvellVsp"),
+	mrvlVspServerInstance = &mrvlVspServer{
```

Comment:
this `New*()` method stores a pointer to the instance in a global variable. This seems a bad name for what it does.

Regardless, I don't understand why this global exists at all. I think it should be dropped, and places that require this instance, should get it passed down as an argument by the caller. The only user of the global is `Listen()`, which already has this instance via the `(vsp *mrvlVspServer)` parameter.

Reply:
Agreed we don't really need to store this in a global. We don't do this elsewhere in the code.